internal/statemachine/pack.go

```diff
@@ -0,0 +1,55 @@
+package statemachine
+
+import (
+	"fmt"
+
+	"github.com/canonical/ubuntu-image/internal/commands"
+)
+
+var packStates = []stateFunc{
+	{"prepare_pack", (*StateMachine).preparePack},
+	{"make_temporary_directories", (*StateMachine).makeTemporaryDirectories},
+	{"populate_temporary_directories", (*StateMachine).populateTemporaryDirectories},
+	{"load_gadget_yaml", (*StateMachine).loadGadgetYaml},
+	{"set_artifact_names", (*StateMachine).setArtifactNames},
```

Comment:
A minor thing, can be in a follow up thing: should we move `setArtifactNames` from `snap_states.go` to `common_states.go` now that we use it for both classic builds and snap builds?
Of course we use a lot of classic states in the pack command, but ideology-wise pack either uses its own, common or classic states.

Reply:
I agree. I hesitated to do it because it changes the meaning of "common" I think. I am OK to do it as long as this is clear that the state functions we have in `common_states.go` are "common" in the sense of "used in at least 2 different builds, and not in the sense of "usable by *every* builds".